cnd/src/btsieve/ethereum/mod.rs

```diff
@@ -192,71 +376,57 @@ fn seen_block_or_predates_start_of_swap(
-{
-    let needs_receipt = pattern.needs_receipts(&block);
-    let block_hash = block
-        .hash
-        .ok_or_else(|| anyhow::anyhow!("block without hash"))?;
+impl Predates for Block {
+    fn predates(&self, timestamp: NaiveDateTime) -> bool {
+        let unix_timestamp = timestamp.timestamp();
 
-    if needs_receipt {
-        tracing::debug!(
-            "bloom-filter of block {:x} suggests to fetch receipts for {:?}",
-            block_hash,
-            pattern
-        );
-    } else {
-        tracing::debug!(
-            "bloom-filter of block {:x} suggests to not fetch receipts for {:?}",
-            block_hash,
-            pattern
-        );
+        self.timestamp < U256::from(unix_timestamp)
     }
+}
 
-    for transaction in block.transactions.into_iter() {
-        let tx_hash = transaction.hash;
-
-        let receipt = connector
-            .receipt_by_hash(tx_hash)
-            .compat()
-            .await?
-            .ok_or_else(|| {
-                anyhow::anyhow!(
-                    "Could not get transaction receipt for transaction {:x}",
-                    tx_hash
-                )
-            })?;
-
-        let result = pattern.matches(&transaction, &receipt);
-
-        tracing::debug!(
-            "matching {:?} against transaction {:x} yielded {}",
-            pattern,
-            tx_hash,
-            result
-        );
-
-        if result {
-            return Ok(Some(TransactionAndReceipt {
-                transaction,
-                receipt,
-            }));
-        }
-    }
+#[derive(Clone, Copy, Default, Eq, PartialEq, serde::Serialize, serdebug::SerDebug)]
+#[serde(transparent)]
+pub struct Topic(pub H256);
 
-    Ok(None)
+/// Event works similar to web3 filters:
+/// https://web3js.readthedocs.io/en/1.0/web3-eth-subscribe.html?highlight=filter#subscribe-logs
+/// E.g. this `Event` would match this `Log`:
+/// ```rust, ignore
+/// Event {
+/// address: "0xe46FB33e4DB653De84cB0E0E8b810A6c4cD39d59",
+/// topics: [
+/// None,
+/// 0x000000000000000000000000e46fb33e4db653de84cb0e0e8b810a6c4cd39d59,
+/// None,
+/// ],
+/// ```
+/// ```rust, ignore
+/// Log:
+/// [ { address: "0xe46FB33e4DB653De84cB0E0E8b810A6c4cD39d59",
+/// data: "0x123",
+/// ..
+/// topics:
+/// [ "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+/// "0x000000000000000000000000e46fb33e4db653de84cb0e0e8b810a6c4cd39d59",
+/// "0x000000000000000000000000d51ecee7414c4445534f74208538683702cbb3e4" ],
+/// },
+/// .. ] //Other data omitted
+/// }
+/// ```
+#[derive(Clone, Default, Eq, PartialEq, serde::Serialize, serdebug::SerDebug)]
+pub struct Event {
+    pub address: Address,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub topics: Vec<Option<Topic>>,
 }
 
-impl Predates for Block {
-    fn predates(&self, timestamp: NaiveDateTime) -> bool {
-        let unix_timestamp = timestamp.timestamp();
+trait TransactionStatusOk {
```

Comment:
This could actually live in the general `ethereum` module.

It would also be nice if the function naming would give away that the fn returns a bool. Something like:

```rust
fn is_status_ok() -> bool {

}
```

Reply:
Fixed as suggested.